include/seqan3/alignment/pairwise/detail/policy_alignment_result_builder.hpp

```diff
@@ -0,0 +1,111 @@
+ *
+ * \details
+ *
+ * Implements the interfaces to build the alignment result based on the previously selected output configurations.
+ */
+template <typename alignment_configuration_t>
+#if !SEQAN3_WORKAROUND_GCC_93467
+//!\cond
+    requires is_type_specialisation_of_v<alignment_configuration_t, configuration>
+//!\endcond
+#endif // !SEQAN3_WORKAROUND_GCC_93467
+class policy_alignment_result_builder
+{
+protected:
+    //!\brief The configuration traits type.
+    using traits_type = alignment_configuration_traits<alignment_configuration_t>;
+    //!\brief The alignment result type.
+    using result_type = typename traits_type::alignment_result_type;
+
+    static_assert(!std::same_as<result_type, empty_type>, "The alignment result type was not configured.");
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    policy_alignment_result_builder() = default; //!< Defaulted.
+    policy_alignment_result_builder(policy_alignment_result_builder const &) = default; //!< Defaulted.
+    policy_alignment_result_builder(policy_alignment_result_builder &&) = default; //!< Defaulted.
+    policy_alignment_result_builder & operator=(policy_alignment_result_builder const &) = default; //!< Defaulted.
+    policy_alignment_result_builder & operator=(policy_alignment_result_builder &&) = default; //!< Defaulted.
+    ~policy_alignment_result_builder() = default; //!< Defaulted.
+
+    /*!\brief Construction and initialisation using the alignment configuration.
+     * \param[in] config The alignment configuration [not used in this context].
+     */
+    policy_alignment_result_builder(alignment_configuration_t const & SEQAN3_DOXYGEN_ONLY(config))
+    {}
+    //!\}
+
+    /*!\brief Builds the seqan3::alignment_result based on the given alignment result type and then invokes the
+     *        given callable with the result.
+     *
+     * \tparam indexed_sequence_pair_t The type of the indexed sequence pair.
+     * \tparam score_t The type of the score.
+     * \tparam callback_t The type of the callback to invoke.
+     *
+     * \param[in] indexed_sequence_pair The indexed sequence pair.
+     * \param[in] score The best alignment score.
+     * \param[in] callback The callback to invoke with the generated result.
+     *
+     * \details
+     *
+     * Generates a seqan3::alignment_result object with the results computed during the alignment. Depending on the
+     * seqan3::align_cfg::result configuration only the requested values are stored. In some cases some additional
+     * work is done to generate the requested result. For example computing the associated alignment from the traceback
+     * matrix.
+     */
+    template <typename indexed_sequence_pair_t, typename score_t, typename callback_t>
+    //!\cond
+        requires std::invocable<callback_t, result_type>
+    //!\endcond
+    void build_result_and_invoke([[maybe_unused]] indexed_sequence_pair_t && indexed_sequence_pair,
```

Comment:
```suggestion
    void make_result_and_invoke([[maybe_unused]] indexed_sequence_pair_t && indexed_sequence_pair,
```

I have a strong preference for make.

Reply:
should the policy then not be called `maker` 🤣

Reply:
No, it is the builder pattern, but we create = make results. BTW in `policy_search_result_builder` we called it `make_results`.